docs/development.rst

```diff
@@ -64,6 +64,15 @@ will override the defaults.
 Server Development
 ------------------
 
+All commits to the core python code must work in all versions of python 2.7 and above.
+Python code in plugins should also work in 2.7 and 3.x, but some plugins may depend
```

Comment:
I'd favor being explicit about what versions we test with with the caveat that we aren't doing anything that should disallow 3.0-3.3.